### ISSUE TITLE
Cap framerate by default to prevent possible GPU damage

### ISSUE THREAD

**Issue by matjaeck** [OpenRA/OpenRA#20557]
## Motivation

Graphics cards and fans can fail over time if a static or low-detail scene is rendered at an uncapped rate. See https://web.archive.org/web/20170903085237/http://www.dailytech.com/Hot+Starcraft+II+is+Frying+Graphics+Cards+Blizzard+Issues+Temporary+Fix/article19224.htm and https://games.slashdot.org/story/10/08/02/1251203/is-starcraft-ii-killing-graphics-cards. Although it is not primarily the application's job to prevent the hardware from overheating and breaking fans, the application should use the available resources wisely and not just push the limits.

## Proposed solution

- Cap framerate in static scenes at for example 10 frames per second. This saves energy and prevents that the frame rate skyrockets, for example in the main menu screen when the background screen is "paused". It is a common thing in games to cap frame rate in loading or other mostly static scenes.

- Universally cap framerate at a reasonable refresh rate, for example at 240 frames per second which is the maximum you can set via the "Frame Limiter" setting.

**Comment by Mailaender** [comment]
We already do. The static limiter was just replaced with VSync in https://github.com/OpenRA/OpenRA/pull/17439.

**Comment by PunkPun** [comment]
No, we don't do this. We don't automatically reduce framerate nor we let it be set lower than 25

**Comment by Mailaender** [comment]
https://github.com/OpenRA/OpenRA/blob/b619dd14c56767425aabf9f3101ba1f8d0c551e1/OpenRA.Game/Settings.cs#L176-L177

**Comment by penev92** [comment]
By the way... while @Mailaender is correct and we do have VSync on by default (which is a bit weird but okay...), if we look closely at the original post, it requests/suggests 2 things, both pretty reasonable (especially as a bonus to VSync being on by default), neither of which we do currently.

**Comment by pchote** [comment]
Also note that VSync doesn't work everywhere.

**Comment by Mailaender** [comment]
So the plan is to revert fe13196f028cc0ecde4799b431537c2fa116a043 then?

**Comment by penev92** [comment]
No, the plan would be to:
1. Add serious frame limiting to static screens (drop it to 1? to 10?) regardless of user settings because it makes little sense to draw static screens at 240FPS.
2. Add an implicit limit at say 240 or 360 FPS. Currently disabling all limiters can result in FPS reaching 1000 (on my machine), which can't be good for the hardware and is not something that brings any value. *Just in case we can have an advanced option to explicitly disable that for the 1-2 people that might want to run an experiment.*

**Comment by Mailaender** [comment]
I can confirm that VSync is not reliable: https://github.com/OpenRA/OpenRA/issues/20681

**Comment by PunkPun** [comment]
We should probably mention that lower framerates also means player input is less responsive. I saw games like SYNTHETIK add this disclaimer
